internal/extensions/extensions.go

```diff
@@ -0,0 +1,219 @@
+package extensions
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var internalExtensionRegex = regexp.MustCompile(`^internal:[a-z_0-9]+$`)
+var externalExtensionRegex = regexp.MustCompile(`^[a-z_0-9]+$`)
+
+type invalidInternalExtensionErr struct {
+	invalidExt string
+}
+
+func (e *invalidInternalExtensionErr) Error() string {
+	return fmt.Sprintf("Internal extension format is invalid (%v). It should be in the format `internal:<extension>` with only lowercase letters and underscores", e.invalidExt)
+}
+
+type invalidExternalExtensionErr struct {
+	invalidExt string
+}
+
+func (e *invalidExternalExtensionErr) Error() string {
+	return fmt.Sprintf("External extension format is invalid (%v). It should be in the format `<extension>` with only lowercase letters and underscores", e.invalidExt)
+}
+
+// Extensions represents a registry of extensions.
+type Extensions struct {
+	// internal extensions are related to MicroCluster capabilities, not capabilities of
+	// MicroCluster-backed services.
+	//
+	// To distinguish between internal and external extensions, internal extensions
+	// are prefixed with a `internal:` value.
+	// e.g, `internal:runtime_extension_v1`, `internal:runtime_extension_v2`, ...
+	internal map[string]struct{}
+
+	// External extensions are related to capabilities of MicroCluster-backed services.
+	// e.g, `microovn_custom_encapsulation_ip`
+	external map[string]struct{}
+}
```

Comment:
I think having a structure like this might be more than is required. In fact, it also loses information about the order that the extensions occurred (not sure if that matters). Since internal extensions are prefixed with `internal:`, can we just have `type Extensions []string`?

Reply:
fixed